.maestro/setup_ui_tests.sh

```diff
@@ -0,0 +1,120 @@
+#!/bin/zsh
+
+### Set up environment for UI testing
+
+source .maestro/common.sh
+
+## Constants
+
+# The simulator command requires the hyphens
+target_device="iPhone-15"
+target_os="iOS-17-2"
+
+## Functions
+
+check_maestro() {
+
+    local command_name="maestro"
+    local known_version="1.36.0"
+
+    if command -v $command_name > /dev/null 2>&1; then
+      local version_output=$($command_name -v 2>&1)
+      
+      local command_version=$(echo $version_output | grep -oE '[0-9]+\.[0-9]+\.[0-9]+')
+
+      if [[ $command_version == $known_version ]]; then
+        echo "ℹ️ maestro version matches: $command_version"
+      else
+        echo "‼️ maestro version does not match. Expected: $known_version, Got: $command_version"
+        exit 1
+      fi
+    else
+      echo "‼️ maestro not found install using the following commands:"
+      echo
+      echo "curl -Ls \"https://get.maestro.mobile.dev\" | bash"
+      echo "brew tap facebook/fb"
+      echo "brew install facebook/fb/idb-companion"
+      echo
+      exit 1
+    fi
+}
+
+build_app() {
+
+    if [ -d "$derived_data_path" ]; then
+        echo "⚠️ Removing previously created $derived_data_path"
+        rm -rf $derived_data_path
+    fi
+
+    echo "⏲️ Building the app"
+    set -o pipefail && xcodebuild -scheme "DuckDuckGo" -destination "platform=iOS Simulator,name=iPhone 15,OS=17.2" -derivedDataPath "$derived_data_path" -skipPackagePluginValidation -skipMacroValidation ONLY_ACTIVE_ARCH=NO | tee xcodebuild.log
+    if [ $? -ne 0 ]; then
+        echo "‼️ Unable to build app into $derived_data_path"
+        exit 1
+    fi
+
+}
+
+## Main Script
+
+check_is_root
```

Comment:
This can't be invoked unless the "source" operation above is successful, but it will be successful only if the script is invoked from the root folder (otherwise ".maestro/common.sh" will be invalid). 
So, if invoked from other folders, the script will print an error message that this command is unknown, but it will continue and try to run the next instructions even if it's on the wrong folder. 

An option could be to use dirname "$0" to get the path of this file and use it in the source command instead of the static path.

Reply:
Basically made it so you can run these scripts from anywhere so this is fixed.